src/notification.c

```diff
@@ -382,6 +363,122 @@ check_notification (GVariant *notification,
   return TRUE;
 }
 
+static gboolean
+validate_icon_more (GVariant *v)
+{
+  g_autoptr(GIcon) icon = g_icon_deserialize (v);
+  GBytes *bytes;
+  g_autofree char *name = NULL;
+  const char *argv[] = { LIBEXECDIR "/xdg-desktop-portal-validate-icon", "name", NULL };
+  xdp_autofd int fd = -1;
+  g_autoptr(GOutputStream) stream = NULL;
+  gssize written;
+  int status;
+  g_autofree char *err = NULL;
+  g_autoptr(GError) error = NULL;
+
+  if (G_IS_THEMED_ICON (icon))
+    {
+      g_autofree char *a = g_strjoinv (" ", (char **)g_themed_icon_get_names (G_THEMED_ICON (icon)));
+      g_debug ("Icon validation: themed icon (%s) is ok", a);
+      return TRUE;
+    }
+
+  if (!G_IS_BYTES_ICON (icon))
+    {
+      g_warning ("Unexpected icon type: %s", G_OBJECT_TYPE_NAME (icon));
```

Comment:
I think we can support `GFileIcon` just fine. We would just want the validator to spit out a validated file on the host or something.

Reply:
Our stance on passing files to portals so far has been: everything must be passed as fd. If we accept a path, we have to worry about validating that it doesn't point to some host location.

Reply:
Pointing to a host location is safe though because this is only consumed by the host itself.

Reply:
it is still a possible information leak - it can cause your secret photos to appear in a notification on your screen, if the app figures out the right filename to pass...

Reply:
Hmm, alright.